### DISCUSSION TITLE
Contribute

### DISCUSSION THREAD

**Discussion by adynemo** [maxpozdeev/mytinytodo#66] [General]
Hello!
If I want to contribute at this project to develop some feature, have you a process?
I have some ideas like a different theme, better responsive, more security with config data into database and the possibility for several users to have an account.

Thks :)

**Comment by maxpozdeev** [comment]
Hi!
You can look at https://github.com/alex-LE/yourTinyTodo

As for me I want to implement in new version:
- ✓ user configuration in database (db type and connection in separate readonly file)
- ✓ language files convert to json
- ✓ markdown support in task note
- multiple users

But no ETA :(

**Comment by ghost** [comment]
I am also interested in contributing to this project.  Are you referring contributors to a different fork?  I see there are 14 forks of this project and just want to make sure I'm contributing to the best one.  I run your version at home and love it, but wish it had PDO support.  Multi user support would be important too, and I'm willing to work on both. Please advise...

**Comment by maxpozdeev** [comment]
The first big and old fork is yourTinyTodo. Seems it is not active now, but you may ask its developer. I do not know about implemented features in newer forks.

**Comment by adynemo** [comment]
This repo (https://github.com/alex-LE/yourTinyTodo) implements some interesting features, but it is too old. The last commit was 6 years ago.
Do you want to have some contributing?

**Comment by maxpozdeev** [comment]
My point of view - I don't want to make the project too big and complicated. But some features I will accept. So if you plan to create something huge then its better to create new repo :)

New theme is good idea, especially for mobiles. But much of interface code is produced in js. Here is the need to find the way to separate interface and logics.

**Comment by adynemo** [comment]
Ok I see. I will fork your project ASAP for those features.
Thks for your work :)

**Comment by ghost** [comment]
My family and I use this application for shared shopping lists, so it is frequently used on mobile devices in our use case.  I think having a mobile interface would be huge for a lot of people, and may make this more useful and accessible.

I am relatively new to collaborating on projects like this with git, so pardon my ignorance- If there are many different forks of the project, wouldn't end users have to decide what fork they wanted to install, leaving them with the problem of choosing between features implemented in different forks?  Say if fork A has great mobile responsiveness but fork B has PDO and support for multiple users, users couldn't easily have a version with both features unless both of the features were built into one fork, right?

**Comment by maxpozdeev** [comment]
> Say if fork A has great mobile responsiveness but fork B has PDO and support for multiple users, users couldn't easily have a version with both features unless both of the features were built into one fork, right?

Right.

**Comment by ghost** [comment]
I'm not planning on doing anything huge I don't think; just want to convert MySQL queries to PDO queries, maybe add some more language support, and eventually add multi user support.  It looks like all three of us have some thoughts to improve this application, so it makes sense to me to all work on one repository since we're all working on different features that would enhance the overall usability and reach of this application, but maybe I'm just missing something.  Thoughts?

**Comment by adynemo** [comment]
So... What are we doing?
@maxpozdeev can we contribute to your project or do you prefer we do a fork?
I'm agree with @jmarmorato to contribute instead of make the 15th fork but it's up to you.
